test/e2e/sagemaker/tests/test_hpo.py

```diff
@@ -0,0 +1,151 @@
+"""Integration tests for the SageMaker HyperParameterTuning API.
+"""
+
+import boto3
+import pytest
+import logging
+from typing import Dict
+import time
+
+from sagemaker import SERVICE_NAME, service_marker, CRD_GROUP, CRD_VERSION
+from sagemaker.replacement_values import REPLACEMENT_VALUES
+from common.resources import load_resource_file, random_suffix_name
+from common import k8s
+
+RESOURCE_PLURAL = 'hyperparametertuningjobs'
+
+
+@pytest.fixture(scope="module")
+def sagemaker_client():
+    return boto3.client('sagemaker')
+
+
+@pytest.fixture(scope="module")
+def xgboost_hpojob():
+    resource_name = random_suffix_name("xgboost-hpojob", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["HPO_JOB_NAME"] = resource_name
+
+    hpojob = load_resource_file(
+        SERVICE_NAME, "xgboost_hpojob", additional_replacements=replacements)
+    logging.debug(hpojob)
+
+    # Create the k8s resource
+    reference = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL, resource_name, namespace="default")
+    resource = k8s.create_custom_resource(reference, hpojob)
+    resource = k8s.wait_resource_consumed_by_controller(reference)
+
+    assert resource is not None
+
+    yield (reference, resource)    
+
+    # Delete the k8s resource if not already deleted by tests
+    try:
+        k8s.delete_custom_resource(reference)
+    except:
+        pass
+
+
+@service_marker
+@pytest.mark.canary
+class TestHPO:
+    def _get_created_hpo_job_status_list(self):
+        return ["InProgress", "Completed"]
+
+    def _get_stopped_hpo_job_status_list(self):
+        return ["Stopped", "Stopping"]
+
+    def _get_sagemaker_hpo_job_arn(self, sagemaker_client, hpo_job_name: str):
+        try:
+            hpo_desc = sagemaker_client.describe_hyper_parameter_tuning_job(
+                HyperParameterTuningJobName=hpo_job_name
+            )
+            return hpo_desc["HyperParameterTuningJobArn"]
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find an hpo job with the name {hpo_job_name}"
+            )
+            return None
+
+    def _get_sagemaker_hpo_job_status(
+        self, sagemaker_client, hpo_job_name: str
+    ):
+        try:
+            hpo_job = sagemaker_client.describe_hyper_parameter_tuning_job(
+                HyperParameterTuningJobName=hpo_job_name
+            )
+            return hpo_job["HyperParameterTuningJobStatus"]
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find an hpo job with the name {hpo_job_name}"
+            )
+            return None
+
+    def test_create_hpo(self, xgboost_hpojob):
+        (reference, resource) = xgboost_hpojob
+        assert k8s.get_resource_exists(reference)
+
+    def test_hpo_has_correct_arn(self, sagemaker_client, xgboost_hpojob):
+        (reference, _) = xgboost_hpojob
+        resource = k8s.get_resource(reference)
```

Comment:
dont need to get_resource here, see https://github.com/aws-controllers-k8s/community/pull/706#discussion_r591941021

same for other places where you need only spec fields

Reply:
fixed